cmake/emu/nsim.cmake

```diff
@@ -1,46 +1,54 @@
 # SPDX-License-Identifier: Apache-2.0
-if((${CONFIG_SOC_NSIM_HS_SMP}) OR (${CONFIG_SOC_NSIM_HS6X_SMP}))
+if("${BOARD_DEBUG_RUNNER}" STREQUAL "mdb-nsim" OR "${BOARD_FLASH_RUNNER}" STREQUAL "mdb-nsim")
 # mdb is required to run nsim multicore targets
 find_program(
   MDB
   mdb
   )
+set(MDB_BASIC_OPTIONS -nooptions -nogoifmain -toggle=include_local_symbols=1)
 
-if(${CONFIG_SOC_NSIM_HS_SMP})
- set(MDB_ARGS mdb_hs_smp.args)
-elseif(${CONFIG_SOC_NSIM_HS6X_SMP})
- set(MDB_ARGS mdb_hs6x_smp.args)
+# remove previous .sc.project folder which has temporary settings for MDB.
+set(MDB_OPTIONS ${CMAKE_COMMAND} -E rm -rf ${APPLICATION_BINARY_DIR}/.sc.project)
+if(CONFIG_MP_NUM_CPUS GREATER 1)
+  set(MULTIFILES ${MDB} -multifiles=)
+  foreach(val RANGE ${CONFIG_MP_NUM_CPUS})
+    if(val LESS CONFIG_MP_NUM_CPUS)
+      MATH(EXPR PSET_NUM "${CONFIG_MP_NUM_CPUS}-${val}")
+      MATH(EXPR CORE_NUM "${CONFIG_MP_NUM_CPUS}-${val}-1")
+      if (PSET_NUM GREATER 0)
```

Comment:
As I remember we should shouldn't add extra space between `if` and `(` in cmake.